### ISSUE TITLE
genders file to include other files

### ISSUE THREAD

**Issue by stuartm-dugeo** [chaos/genders#23]
Would be great to have the ability for nested genders files... 

eg. /etc/genders to be

    include /etc/genders_c1
    include /etc/genders_c2

    name1 attr1,attr2=a

This would allow us to have 1 genders file manage multiple clusters.

**Comment by BenCasses** [comment]
I hear that this concept was discussed early in the creation of genders.  It's a good idea that would at least cut down on mistakes.  Pushback at the time, reasonably, was: "feature-creep".  I'm going to add this to the TODO list and pick it up when I have cycles.

**Comment by stuartm-dugeo** [comment]
Thanks.

**Comment by chu11** [comment]
One comment/thing to ponder, if an old version of genders parsed

```
include /etc/genders_c1
```

I think it would see a node named "include" and an attribute named "/etc/genders_c1".

Perhaps it should be:

```
#include /etc/genders_c1
```

So atleast the include directive is not read b/c its a comment in earlier versions.

Or are we just introducing an alternate unexpected behavior so it doesn't matter either way?

**Comment by garlick** [comment]
Least surprise might be to have the new include directive cause a parse error in old genders versions, if that's even possible - looks like we are pretty liberal about what constitutes a valid line of input.

**Comment by BenCasses** [comment]
I agree that it should fail visibly when using an older version of genders.  Whoever used the include probably did it on purpose and should have a strong indication of badness rather than some genders silently disappearing if include is interpreted as a comment.   Seems like a 3-word line breaks the parser.  Maybe something like 
```
include gender /etc/genders_c1
```

**Comment by garlick** [comment]
I'm not sure there is a way to make this pretty, but would it be useful to introduce a special directive prefix character so that future versions of the library can add directives beyond "include", and unknown directives can be detected and called out as such?

For example, maybe use
```
% include /etc/genders_c1
```
then in new parser
```
%include /etc/genders_c1
Line 1: % directive must be followed by white space

% foo bar
Line 1: % foo is an unknown directive
```
as opposed to
```
include /etc/genders_c1
(not detected)

include genders /etc/gendres_c1
Line 1: white space in attribute list
```

**Comment by baallan** [comment]
I've had good results (and more easily debugged results) just concatenating various input pieces with  application logic into a singletmp  input file before parsing genders.

**Comment by baallan** [comment]
If you go with "include genders /etc/gendres_c1" syntax, for completeness consider
include genders <file>
include pipe <program> [other args]
where the second reads the output of system() calling program with the args.
